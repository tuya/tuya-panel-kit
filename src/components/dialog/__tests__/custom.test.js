import React from 'react';
import renderer from 'react-test-renderer';
import { View, Text } from 'react-native';
import Custom from '../custom';

describe('Custom Component', () => {
  it('basic render', () => {
    const component = renderer.create(
      <Custom
        title="标题"
        subTitle="副标题"
        confirmText="确认"
        cancelText="取消"
        onConfirm={jest.fn((value, { close: callback }) => callback())}
        onCancel={jest.fn()}
        content={
          <View
            style={{
              height: 300,
              alignItems: 'center',
              justifyContent: 'center',
            }}
          >
            <Text style={{ fontSize: 32, color: '#000' }}>自定义内容</Text>
          </View>
        }
      />
    );
    const instance = component.getInstance();
    instance._handleConfirm();
    instance._handleCancel();
    instance._handleHide();
    expect(component).toMatchSnapshot();
  });

  it(' render with footer', () => {
    const component = renderer.create(
      <Custom
        title="标题"
        subTitle="副标题"
        confirmText="确认"
        cancelText="取消"
        onConfirm={jest.fn(() => console.log('Hello'))}
        footer={
          <View
            style={{
              height: 300,
              alignItems: 'center',
              justifyContent: 'center',
            }}
          >
            <Text style={{ fontSize: 32, color: '#000' }}>自定义Footer</Text>
          </View>
        }
        content={
          <View
            style={{
              height: 300,
              alignItems: 'center',
              justifyContent: 'center',
            }}
          >
            <Text style={{ fontSize: 32, color: '#000' }}>自定义内容</Text>
          </View>
        }
      />
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  it(' render with motionType', () => {
    const component = renderer.create(
      <Custom
        title="标题"
        subTitle="副标题"
        confirmText="确认"
        cancelText="取消"
        onConfirm={jest.fn(() => console.log('Hello'))}
        motionType="none"
        content={
          <View
            style={{
              height: 300,
              alignItems: 'center',
              justifyContent: 'center',
            }}
          >
            <Text style={{ fontSize: 32, color: '#000' }}>自定义内容</Text>
          </View>
        }
      />
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});
