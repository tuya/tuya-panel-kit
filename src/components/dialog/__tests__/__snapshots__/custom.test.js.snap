// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Custom Component  render with footer 1`] = `
<View
  renderToHardwareTextureAndroid={false}
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "center",
      "opacity": 0,
      "transform": Array [
        Object {
          "scale": 0.8,
        },
      ],
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "backgroundColor": "#fff",
          "borderRadius": 32,
          "justifyContent": "center",
          "overflow": "hidden",
          "width": 622,
        },
        null,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "alignSelf": "stretch",
            "borderBottomColor": "rgba(0, 0, 0, 0.1)",
            "borderBottomWidth": 0.5,
            "height": 55,
            "justifyContent": "center",
          },
          null,
        ]
      }
    >
      <Text
        accessible={true}
        align={null}
        allowFontScaling={false}
        color={null}
        ellipsizeMode="tail"
        numberOfLines={2}
        size={17}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "color": "#333",
              "fontSize": 17,
            },
            Array [
              Object {
                "color": "rgba(0, 0, 0, 0.9)",
                "flexWrap": "wrap",
                "fontSize": 17,
                "fontWeight": "500",
                "textAlign": "center",
              },
              null,
            ],
          ]
        }
        type={null}
        weight={null}
      >
        标题
      </Text>
      <Text
        accessible={true}
        align={null}
        allowFontScaling={false}
        color={null}
        ellipsizeMode="tail"
        size={15}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "color": "#333",
              "fontSize": 15,
            },
            Array [
              Object {
                "color": "rgba(0, 0, 0, 0.5)",
                "flexWrap": "wrap",
                "fontSize": 15,
                "textAlign": "center",
              },
              null,
            ],
          ]
        }
        type={null}
        weight={null}
      >
        副标题
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "height": 300,
          "justifyContent": "center",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#000",
            "fontSize": 32,
          }
        }
      >
        自定义内容
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "height": 300,
          "justifyContent": "center",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#000",
            "fontSize": 32,
          }
        }
      >
        自定义Footer
      </Text>
    </View>
  </View>
</View>
`;

exports[`Custom Component  render with motionType 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "alignSelf": "center",
        "backgroundColor": "#fff",
        "borderRadius": 32,
        "justifyContent": "center",
        "overflow": "hidden",
        "width": 622,
      },
      null,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "alignSelf": "stretch",
          "borderBottomColor": "rgba(0, 0, 0, 0.1)",
          "borderBottomWidth": 0.5,
          "height": 55,
          "justifyContent": "center",
        },
        null,
      ]
    }
  >
    <Text
      accessible={true}
      align={null}
      allowFontScaling={false}
      color={null}
      ellipsizeMode="tail"
      numberOfLines={2}
      size={17}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "color": "#333",
            "fontSize": 17,
          },
          Array [
            Object {
              "color": "rgba(0, 0, 0, 0.9)",
              "flexWrap": "wrap",
              "fontSize": 17,
              "fontWeight": "500",
              "textAlign": "center",
            },
            null,
          ],
        ]
      }
      type={null}
      weight={null}
    >
      标题
    </Text>
    <Text
      accessible={true}
      align={null}
      allowFontScaling={false}
      color={null}
      ellipsizeMode="tail"
      size={15}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "color": "#333",
            "fontSize": 15,
          },
          Array [
            Object {
              "color": "rgba(0, 0, 0, 0.5)",
              "flexWrap": "wrap",
              "fontSize": 15,
              "textAlign": "center",
            },
            null,
          ],
        ]
      }
      type={null}
      weight={null}
    >
      副标题
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "height": 300,
        "justifyContent": "center",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#000",
          "fontSize": 32,
        }
      }
    >
      自定义内容
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "alignSelf": "stretch",
          "borderTopColor": "rgba(0, 0, 0, 0.1)",
          "borderTopWidth": 0.5,
          "flexDirection": "row",
          "height": 55,
        },
        null,
      ]
    }
  >
    <View
      accessibilityLabel="Dialog.Cancel"
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "borderRightColor": "rgba(0, 0, 0, 0.1)",
          "borderRightWidth": 0.5,
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "center",
          "opacity": 1,
        }
      }
    >
      <Text
        accessible={true}
        align={null}
        allowFontScaling={false}
        color={null}
        ellipsizeMode="tail"
        size={17}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "color": "#333",
              "fontSize": 17,
            },
            Array [
              Object {
                "color": "rgba(0, 0, 0, 0.7)",
                "fontSize": 17,
                "textAlign": "center",
              },
              null,
            ],
          ]
        }
        type={null}
        weight={null}
      >
        取消
      </Text>
    </View>
    <View
      accessibilityLabel="Dialog.Confirm"
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "borderRightColor": "rgba(0, 0, 0, 0.1)",
          "borderRightWidth": 0,
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "center",
          "opacity": 1,
        }
      }
    >
      <Text
        accessible={true}
        align={null}
        allowFontScaling={false}
        color={null}
        ellipsizeMode="tail"
        size={17}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "color": "#333",
              "fontSize": 17,
            },
            Array [
              Object {
                "color": "#FF4800",
                "fontSize": 17,
                "fontWeight": "bold",
                "textAlign": "center",
              },
              null,
            ],
          ]
        }
        type={null}
        weight={null}
      >
        确认
      </Text>
    </View>
  </View>
</View>
`;

exports[`Custom Component basic render 1`] = `
<View
  renderToHardwareTextureAndroid={true}
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "center",
      "opacity": 0,
      "transform": Array [
        Object {
          "scale": 0.8,
        },
      ],
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "backgroundColor": "#fff",
          "borderRadius": 32,
          "justifyContent": "center",
          "overflow": "hidden",
          "width": 622,
        },
        null,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "alignSelf": "stretch",
            "borderBottomColor": "rgba(0, 0, 0, 0.1)",
            "borderBottomWidth": 0.5,
            "height": 55,
            "justifyContent": "center",
          },
          null,
        ]
      }
    >
      <Text
        accessible={true}
        align={null}
        allowFontScaling={false}
        color={null}
        ellipsizeMode="tail"
        numberOfLines={2}
        size={17}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "color": "#333",
              "fontSize": 17,
            },
            Array [
              Object {
                "color": "rgba(0, 0, 0, 0.9)",
                "flexWrap": "wrap",
                "fontSize": 17,
                "fontWeight": "500",
                "textAlign": "center",
              },
              null,
            ],
          ]
        }
        type={null}
        weight={null}
      >
        标题
      </Text>
      <Text
        accessible={true}
        align={null}
        allowFontScaling={false}
        color={null}
        ellipsizeMode="tail"
        size={15}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "color": "#333",
              "fontSize": 15,
            },
            Array [
              Object {
                "color": "rgba(0, 0, 0, 0.5)",
                "flexWrap": "wrap",
                "fontSize": 15,
                "textAlign": "center",
              },
              null,
            ],
          ]
        }
        type={null}
        weight={null}
      >
        副标题
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "height": 300,
          "justifyContent": "center",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#000",
            "fontSize": 32,
          }
        }
      >
        自定义内容
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "alignSelf": "stretch",
            "borderTopColor": "rgba(0, 0, 0, 0.1)",
            "borderTopWidth": 0.5,
            "flexDirection": "row",
            "height": 55,
          },
          null,
        ]
      }
    >
      <View
        accessibilityLabel="Dialog.Cancel"
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#fff",
            "borderRightColor": "rgba(0, 0, 0, 0.1)",
            "borderRightWidth": 0.5,
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "justifyContent": "center",
            "opacity": 1,
          }
        }
      >
        <Text
          accessible={true}
          align={null}
          allowFontScaling={false}
          color={null}
          ellipsizeMode="tail"
          size={17}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "color": "#333",
                "fontSize": 17,
              },
              Array [
                Object {
                  "color": "rgba(0, 0, 0, 0.7)",
                  "fontSize": 17,
                  "textAlign": "center",
                },
                null,
              ],
            ]
          }
          type={null}
          weight={null}
        >
          取消
        </Text>
      </View>
      <View
        accessibilityLabel="Dialog.Confirm"
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#fff",
            "borderRightColor": "rgba(0, 0, 0, 0.1)",
            "borderRightWidth": 0,
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "justifyContent": "center",
            "opacity": 1,
          }
        }
      >
        <Text
          accessible={true}
          align={null}
          allowFontScaling={false}
          color={null}
          ellipsizeMode="tail"
          size={17}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "color": "#333",
                "fontSize": 17,
              },
              Array [
                Object {
                  "color": "#FF4800",
                  "fontSize": 17,
                  "fontWeight": "bold",
                  "textAlign": "center",
                },
                null,
              ],
            ]
          }
          type={null}
          weight={null}
        >
          确认
        </Text>
      </View>
    </View>
  </View>
</View>
`;
